src/main/java/joeduck/JoeDuck.java

```diff
@@ -0,0 +1,186 @@
+                    tasks.removeTask(targetIndex);
+                    storage.writeList(tasks.getTaskList());
+                    return ui.printResponse("Removed " + targetTask);
+                }
+                case "find": {
+                    String keyword = currCommand.args();
+                    return ui.printResponse(tasks.findTask(keyword));
+                }
+                case "todo":
+                    String todoString = currCommand.args();
+                    if (todoString.isEmpty()) {
+                        throw new InvalidCommandException("Todo requires a description.");
+                    }
+                    Todo t = new Todo(todoString);
+                    tasks.addTask(t);
+                    storage.writeList(tasks.getTaskList());
+                    return ui.printResponse("Added Todo:\n" + t);
+                case "deadline": {
+                    String deadlineString = currCommand.args();
+                    String pattern = "(.+) /by (\\d{4}-\\d{2}-\\d{2}+) (\\d{2}:\\d{2})";
+                    Pattern pd = Pattern.compile(pattern);
+                    Deadline d = getDeadline(pd, deadlineString);
+                    tasks.addTask(d);
+                    storage.writeList(tasks.getTaskList());
+                    return ui.printResponse("Added Deadline:\n" + d);
+                }
+                case "event": {
+                    String deadlineString = currCommand.args();
+                    String pattern = "(.+) /from (\\d{4}-\\d{2}-\\d{2}) (\\d{2}:\\d{2}) " +
+                            "/to (\\d{4}-\\d{2}-\\d{2}) (\\d{2}:\\d{2})";
+                    Pattern pe = Pattern.compile(pattern);
+                    Event e = getEvent(pe, deadlineString);
+                    tasks.addTask(e);
+                    storage.writeList(tasks.getTaskList());
+                    return ui.printResponse("Added Event:\n" + e);
+                }
+                default:
+                    throw new InvalidCommandException("Invalid command!");
+            }
+        } catch (JoeDuckException e) {
+            return ui.printError(e.getMessage());
+        } catch (FileNotFoundException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private static Event getEvent(Pattern pe, String deadlineString) throws RegexMatchFailureException {
+        Matcher me = pe.matcher(deadlineString);
+        if (!me.find()) {
+            throw new RegexMatchFailureException("Arguments for creating event is incorrect");
+        }
+
+        String desc = me.group(1);
+        String startDate = me.group(2);
+        LocalDate d1 = LocalDate.parse(startDate);
+        String startTime = me.group(3);
+        LocalTime t1 = LocalTime.parse(startTime);
+        String endDate = me.group(4);
+        LocalDate d2 = LocalDate.parse(endDate);
+        String endTime = me.group(5);
+        LocalTime t2 = LocalTime.parse(endTime);
```

Comment:
The names `d1`, `t1`, etc are not very descriptive, but since `startDate`, etc are already used, perhaps can consider changing `startDate` → `startDateString` and `d1 `→ `startDate`, etc instead?